cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultAppConfigurer.java

```diff
@@ -194,6 +199,13 @@ public void addSchedule(Schedule schedule, SchedulableProgramType programType, S
     schedules.put(schedule.getName(), spec);
   }
 
+  @Override
+  public ScheduleConfigurer scheduleWorkflow(String scheduleName, String workflowName) {
+    AtomicReference<ScheduleCreationSpec> reference = new AtomicReference<>();
```

Comment:
Seems a bit overly complicated. You can just pass the `programSchedules` map to the `DefaultScheduleConfigurer`.